### ISSUE TITLE
incompatible with the lastest ktx

### ISSUE THREAD

**Issue by engineer1109** [SaschaWillems/Vulkan#719]
../external/libktx.a(texture.c.o)：In function‘ktxTexture_createFromStream’：
texture.c:(.text+0x826)：‘ktxTexture1_constructFromStreamAndHeader’undefined reference
texture.c:(.text+0x882)：‘ktxTexture2_constructFromStreamAndHeader’undefined reference

I suggest include the part ktx code you need in your project, not in the git submodule.

**Comment by SaschaWillems** [comment]
Can you elaborate? Did you update the submodules? If so to what version?

**Comment by engineer1109** [comment]
The master branch of https://github.com/KhronosGroup/KTX-Software
I had git submodule --init update

**Comment by engineer1109** [comment]
https://github.com/KhronosGroup/KTX-Software/blob/master/lib/texture.c

Line 342 
https://github.com/KhronosGroup/KTX-Software/blob/master/lib/texture.c
```
 result = ktxTexture1_constructFromStreamAndHeader(tex1, pStream,
                                                          &header.ktx,
                                                          createFlags);
```
It is the function of texture1.c

```
result = ktxTexture2_constructFromStreamAndHeader(tex2, pStream,
                                                          &header.ktx2,
                                                          createFlags);
```
It the the function of texture2.c

**Comment by SaschaWillems** [comment]
What's the commit hash of the updated submodule? I just updated the ktx submodule and my project compiles fine.

Besides those are functions only internally used by ktx, so I don't know why they should fail while building my project.

Did you do a clean build?

**Comment by SaschaWillems** [comment]
It looks like they deprecated some functions. Will check and fix.

**Comment by SaschaWillems** [comment]
I have (tried to) fix the submodule to the latest 3.0 tag. Can you do a clean checkout and try again? For me that now fetches v.3.0.1 of ktx which compiles fine.

**Comment by engineer1109** [comment]
solved
